Source/AwsCommonRuntimeKit/auth/imds/CRTIMDSClient.swift

```diff
@@ -27,11 +27,12 @@ public class CRTIMDSClient {
     /// - Parameters:
     ///    - resourcePath: `String` path of the resource to query
     public func getResource(resourcePath: String) async throws -> String? {
-        let resourcePathCursor = AWSStringByteCursor(resourcePath, allocator: allocator)
         return try await withCheckedThrowingContinuation { (continuation: ResourceContinuation) in
-            let callbackData = CRTIMDSClientResourceCallbackData(continuation: continuation)
-            let pointer: UnsafeMutableRawPointer = fromPointer(ptr: callbackData)
-            aws_imds_client_get_resource_async(rawValue, resourcePathCursor.byteCursor, resourceCallback, pointer)
+            resourcePath.withByteCursor { resourcePathCursor in
+                let callbackData = CRTIMDSClientResourceCallbackData(continuation: continuation)
+                let pointer: UnsafeMutableRawPointer = fromPointer(ptr: callbackData)
```

Comment:
I don't see anything happening to retain callbackData and prevent it from being cleaned up before the e
completion-callback fires?


shouldn't we be doing:
`let pointer = Unmanaged.passRetained(callbackData).toOpaque()`

instead of that using that weird `fromPointer()` function .. that weirdly allocates a POINTER  on the heap but doesn't seem to do anything to keep the thing it's pointing AT alive

feel free to make this a TODO if this kind of thing needs to be fixed a million times in the codebase